compilers/concrete-compiler/compiler/lib/Runtime/simulation.cpp

```diff
@@ -94,7 +96,11 @@ uint64_t sim_bootstrap_lwe_u64(uint64_t plaintext, uint64_t *tlu_allocated,
   double variance = concrete_cpu_variance_blind_rotate(
       input_lwe_dim, glwe_dim, poly_size, base_log, level, 64,
       mlir::concretelang::optimizer::DEFAULT_FFT_PRECISION, variance_bsk);
-  return out + gaussian_noise(0, variance);
+  out = out + gaussian_noise(0, variance);
```

Comment:
you need to check if the output is signed or not, as signed values use the padding bit. You shoud also check if the out + noise overflow (i.e. check before the adding noise)

Reply:
My thoughts on this is that we have two cases:
1. it's signed, and then the padding bit is allowed, so we don't detect overflows
2. It's unsigned, and we check for overflow (use of the padding bit)
but in both cases, we shouldn't check for the addition with the noise, as it's something unrelated to what the user did. Overflow warnings are for the user to know where their computation happened to be overflowing and fix it maybe, but not for detecting overflow of internal implementations.
Wdyt @BourgerieQuentin ?

Reply:
cc @umut-sahin